theano/gpuarray/subtensor.py

```diff
@@ -1182,3 +1182,37 @@ def infer_shape(self, node, shapes):
             diag_size = T.minimum(dim1, dim2)
         out_shape.append(diag_size)
         return [tuple(out_shape)]
+
+
+class GpuAllocDiag(Subtensor):
+    __props__ = ("offset",)
+
+    def __init__(self, offset=0, axis1=0, axis2=1, view=False):
+        self.offset = offset
```

Comment:
You are taking a bunch of arguments in the init, but ignore all of them besides offset, why?